source/codegen/templates/service_helpers.mako

```diff
@@ -491,8 +491,15 @@ ${initialize_standard_input_param(function_name, parameter)}
 % else:
 <%
   size_field_name = size_sources[-1]
+  size_from_param_name = parameter["determine_size_from"][0] if isinstance(parameter["determine_size_from"], list) else parameter["determine_size_from"]
```

Comment:
Consider: Would it make more sense to assert that determine_size_from is not a list when `len-in-bytes` is set? Linked size params don't necessarily have the same type/size. Unless you need that to work for some case in xnet ...

Reply:
`determine_size_from` is always initialized to list in `metadata_mutation`. I have added check for len of that shouldn't be greater than 1 and raise exception.

I don't understand `determine_size_from` mechanism properly and couldn't find any usage of that tag in `functions.py` to validate my changes against. @astarche can you confirm if this logic is what you had in mind?

Reply:
yeah, that looks good. Sorry for the confusion. `determine_size_from` is populated in a transform. It's sort of like a back-pointer to the size mechanism field (which is from array->size param). It can have multiple elements for "linked arrays" in which case there is logic to make sure that they are all the same length. My assumption was that that logic *did not* work for len-in-bytes (because element size may differ). So, the assert you added should help us catch any case where we try to use linked len-in-bytes and at least make sure that someone works through the codegen.